async/src/main/java/org/odk/collect/async/CoroutineScheduler.kt

```diff
@@ -0,0 +1,55 @@
+package org.odk.collect.async
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import java.util.function.Consumer
+import java.util.function.Supplier
+import kotlin.coroutines.CoroutineContext
+
+open class CoroutineScheduler(private val foregroundContext: CoroutineContext, private val backgroundContext: CoroutineContext) : Scheduler {
```

Comment:
We won't always need deferred tasks so thought I'd split the implementations. I think long term it might make sense to split the interfaces but lets see how it evolves.